h2o-automl/src/main/java/ai/h2o/automl/modeling/DummyStepsProvider.java

```diff
@@ -0,0 +1,143 @@
+                }
+            }
+
+            if (buildSpec.build_control.balance_classes) {
+                params._balance_classes = buildSpec.build_control.balance_classes;
+                params._class_sampling_factors = buildSpec.build_control.class_sampling_factors;
+                params._max_after_balance_size = buildSpec.build_control.max_after_balance_size;
+            }
+
+            params._keep_cross_validation_predictions = true;
+            params._keep_cross_validation_models = buildSpec.build_control.keep_cross_validation_models;
+            params._keep_cross_validation_fold_assignment = buildSpec.build_control.nfolds != 0 && buildSpec.build_control.keep_cross_validation_fold_assignment;
+            params._export_checkpoints_dir = buildSpec.build_control.export_checkpoints_dir;
+
+            return params;
+        }
+
+        private static void makeDummyModel(AutoML aml, Job<DRFModel> job) {
+            DRFModel.DRFParameters params = prepareModelParameters(aml);
+            DRFModel model = new DRFModel(job._result, params, new DRFModel.DRFOutput(new DRF(params)));
+            Frame train = aml.getTrainingFrame();
+            model._output._training_metrics = model
+                    .makeMetricBuilder(new String[0])
+                    .makeModelMetrics(model, train, train, train);
+
+            Frame predictions = train.subframe(new String[]{model._output.responseName(),}).deepCopy("dummy_predictions");
+            predictions.add("probabilities", predictions.vec(0));
+            predictions.add("predict", predictions.vec(0));
+            model._output._cross_validation_holdout_predictions_frame_id = predictions._key;
+
+            DKV.put(predictions);
+            DKV.put(model);
+        }
+
+        static abstract class DummySleepStep extends ModelingStep.ModelStep<DRFModel> {
+            DummySleepStep(String id, int weight, AutoML autoML) {
+                super(Algo.DRF, id, weight, autoML);
+            }
+
+            @Override
+            protected Job<DRFModel> startJob() {
+                //FIXME: This part will use max_runtime_secs provided by aml {
+                double maxRuntimeSecs = aml()
+                        .getBuildSpec()
+                        .build_control
+                        .stopping_criteria
+                        .max_runtime_secs_per_model();
+                // if max_runtime_secs_per_model is not specified (or is negative) try to use max_runtime_secs_per_model
+                if (maxRuntimeSecs <= 0) {
+                    maxRuntimeSecs = aml().getBuildSpec().build_control.stopping_criteria.max_runtime_secs();
+                }
+                // if both max_runtime_secs{,_per_model} are not specified or negative use weight
+                if (maxRuntimeSecs <= 0) {
+                    maxRuntimeSecs = _weight;
+                }
+                maxRuntimeSecs = Math.min(maxRuntimeSecs, aml().timeRemainingMs() / 1e3);
```

Comment:
same here: the whole idea of `startJob` method is to focus ONLY on the parameters specific to that step, i.e. tuning algo parameters.
All the more generic params, stopping criteria, frames and columns, xval and so on are assigned in the parent class to avoid duplication and ensure consistency (and in case of `max_runtime_secs`, to ensure that it actually works).